### PR TITLE
DOC: Add example for ``np.random.default_rng().exponential()``

### PR DESCRIPTION
**Pull request by linus-md**
Adds an example.

### CODE REVIEW

**Comment by a-manpathan** [comment]
As per your last commit @linus-md you have contributed the required example for the "np.random.default_rng().exponential()" function, but it is not showing the same example in the actual docs (on the website). So I am a bit confused Does this  Issue need and contribution or is it closed? Please excuse me if this sounds stupid I am a total beginner at this :)

**Comment by rkern** [comment]
The PR is still open. It has not been merged to `main` yet, so the [latest-development docs](https://numpy.org/devdocs/) won't have it yet. The PR still needs to be reviewed by an existing contributor before it will be merged. It's just a matter of us getting the time to do so.

In the meantime, you can see the continuous integration builds of the docs for this PR in the big list of "checks": it's named `ci/circleci: build artifact`. Click the "Details" link for that item and navigate to the [`Generator.exponential`](https://output.circle-artifacts.com/output/job/c54e262b-4673-4ee1-835c-4e88a6dc47c2/artifacts/0/doc/build/html/reference/random/generated/numpy.random.Generator.exponential.html) page, and you will see the successful results of this PR's contribution.

**Comment by linus-md** [comment]
Hi @a-manpathan, if you want to work on something you shuld look at the `Issues` tab and find something that is open and where no pull request has been created. 😄

File: numpy/random/_generator.pyx [hunk 1]
```diff
@@ -440,19 +440,31 @@ cdef class Generator:
...
+
+        The corresponding distribution can be visualized as follows:
+
+        >>> import matplotlib.pyplot as plt
+        >>> scale, size = 4, 10000
+        >>> rng = np.random.default_rng()
+        >>> sample = rng.exponential(scale=scale, size=size)
+        >>> count, bins, _ = plt.hist(sample, 30, density=True)
+        >>> plt.plot(bins, scale*np.exp(-scale*bins), linewidth=2, color='r')
+        >>> plt.show()
```
Comment (mattip) [inline numpy/random/_generator.pyx:467]: Would using semilogy for [the plot](https://output.circle-artifacts.com/output/job/eaf1fd52-3139-468b-8e32-8f9847f19f0c/artifacts/0/doc/build/html/reference/random/generated/numpy.random.Generator.exponential.html#numpy.random.Generator.exponential) make the plot more informative?

Comment (linus-md) [reply numpy/random/_generator.pyx:467]: Thank you for making me aware of this. The problem was that I was using the non-inverted scale. Should look better now.

**Comment by mattip** [comment]
Thanks @linus-md

**Pull request merged**